### PR TITLE
feat: [SPG-155]: Props to expand project setup menu by default

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

* Project setup menu should be default expanded in the Project side nav
* Increased width of Module card

#### Screenshots

Screenshot 2022-10-26 at 5 06 53 PM

Screenshot 2022-10-26 at 5 08 26 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx [hunk 1]
```diff
@@ -16,16 +16,18 @@ interface NavExpandableProps {
...
   title,
   route,
   children,
   className,
-  withoutBorder = false
+  withoutBorder = false,
+  defaultExpanded = false
 }) => {
-  const [mouseEnter, setMouseEnter] = useState<boolean>(false)
+  const [mouseEnter, setMouseEnter] = useState<boolean>(defaultExpanded)
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx:30]: Don't use mouse enter to store this. 
Instead ignore/disable mouse behaviour when this flag is true.

File: src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx [hunk 2]
```diff
@@ -16,24 +16,26 @@ interface NavExpandableProps {
...
+  defaultExpanded = false
 }) => {
-  const [mouseEnter, setMouseEnter] = useState<boolean>(false)
+  const [isExpanded, setIsExpanded] = useState<boolean>(defaultExpanded)
   const { pathname } = useLocation()
   const isSelected = matchPath(pathname, route)
   const timerRef = React.useRef<number | null>(null)
 
   const handleMouseEvent = (val: boolean): void => {
     if (timerRef.current) window.clearTimeout(timerRef.current)
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx:36]: we should also exit/return early from this function if `defaultExpanded` is true.

File: src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ const NavExpandable: React.FC<React.PropsWithChildren<NavExpandableProps>> = ({
       {!withoutBorder && <div className={css.border} />}
       <Layout.Vertical
         className={cx(className, css.main)}
-        onMouseEnter={() => handleMouseEvent(true)}
+        onMouseEnter={() => !isExpanded && handleMouseEvent(true)}
         onMouseLeave={() => handleMouseEvent(false)}
```
Comment (mayankVermaHarness) [inline src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx:48]: As per Abhinav's comment, we can exit early from `onMouseLeave` as well

**Review by mayankVermaHarness — DISMISSED** [review]

File: src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx (hunk 3 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/NavExpandable/NavExpandable.tsx:48]: isExpanded will always be true whenever onMouseLeave will be called, so there's no point adding the condition

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**